blobit-core/src/main/java/org/blobit/core/cluster/BookKeeperBlobManager.java

```diff
@@ -543,13 +545,14 @@ ExecutorService getCallbacksExecutor() {
         return callbacksExecutor;
     }
 
-    void closeAllActiveWritersForTests() {
+    boolean closeAllActiveWritersForTests() {
         List<BucketWriter> actualWriters = new ArrayList<>(activeWriters.
                 values());
         writers.clear();
-        for (BucketWriter writer : actualWriters) {
+        actualWriters.forEach(writer -> {
             writer.awaitTermination();
-        }
+        });
+        return activeWriters.isEmpty();
```

Comment:
I think this will return false most of the times. The actual removal of the writer from the activeWriters map is performed asynchronously

Reply:
@eolivelli  @diegosalvi can you confirm?

Reply:
I think you are right. I have updated PR.